From 95c86b2e95cd7013a66033471bf7001429282196 Mon Sep 17 00:00:00 2001
From: Anjia Wang <anjiawang@gmail.com>
Date: Fri, 29 Mar 2024 14:09:05 -0700
Subject: [PATCH] fix(CPP): clang/llvm build errors

Use Clang instead because GCC with -g doesn't create a DEBUG build.
Removed -Werror since it triggered multiple compilation errors with Clang.

The upstream project is expected to address this issue in the next release.

Ref: https://github.com/ip7z/7zip/issues/20
---
 CPP/7zip/7zip_gcc.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CPP/7zip/7zip_gcc.mak b/CPP/7zip/7zip_gcc.mak
index 8bf0594..7bb4d3a 100644
--- a/CPP/7zip/7zip_gcc.mak
+++ b/CPP/7zip/7zip_gcc.mak
@@ -24,7 +24,7 @@ PROGPATH_STATIC = $(O)/$(PROG)s
 
 
 ifneq ($(CC), xlc)
-CFLAGS_WARN_WALL = -Werror -Wall -Wextra
+CFLAGS_WARN_WALL = -Wall -Wextra
 endif
 
 # for object file
@@ -34,7 +34,7 @@ CFLAGS_BASE_LIST = -c
 # CFLAGS_BASE_LIST = -S
 CFLAGS_BASE = -O2 $(CFLAGS_BASE_LIST) $(CFLAGS_WARN_WALL) $(CFLAGS_WARN) \
  -DNDEBUG -D_REENTRANT -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
- -fPIC
+ -fPIC -g
 
 FLAGS_FLTO = -ffunction-sections
 FLAGS_FLTO = -flto
-- 
2.44.0

